pilot/pkg/xds/ads.go

```diff
@@ -163,24 +163,20 @@ func (s *DiscoveryServer) receiveThread(con *Connection, reqChannel chan *discov
 	firstReq := true
 	for {
 		req, err := con.stream.Recv()
-		con.mu.RLock()
```

Comment:
I think this is not needed. We can directly use in if block - we are already accessing directly from con in if block